[refactor] 계좌/거래내역 목록 조회 패키지 구조 개선

vayu404 opened the pull request:
## ✅ 연관된 이슈

#33 #20 

## 📝 작업 내용

패키지 추가 및 구조 개선 했습니다.

<img width="664" height="636" alt="image" src="https://github.com/user-attachments/assets/6ba1f447-28b3-4be4-8dd0-4c83071a8fd8" />

Commit:
[refactor]계좌/거래내역 조회 패키지 구조 개선

Commit:
[refactor] 거래내역목록조회 코드 개선

vayu404 commented:
/gemini review

vayu404 commented on src/main/java/dev/woori/wooriLearn/domain/edubankapi/eduaccount/repository/EdubankapiTransactionHistoryRepository.java:
```diff
@@ -0,0 +1,31 @@
+            Long accountId,
+            LocalDateTime startDate,
+            LocalDateTime endDate
+    );
```
이 부분은 @query를 통해 메서드 이름을 간결하게 할지 아니면 그냥 이어갈지 뭐가 더 좋을지 모르겠습니다.

vayu404 commented on src/main/java/dev/woori/wooriLearn/domain/edubankapi/eduaccount/service/EdubankapiAccountService.java:
```diff
@@ -0,0 +1,93 @@
+                    if ("WITHDRAW".equalsIgnoreCase(type)) return h.getAmount() < 0;
+                    return true;
+                })
+                .limit(30)
```
교육용 시나리오에서는 많은 거래내역이 있지 않아서 그냥 코드 수정하지 않고 유지하겠습니다.

Commit:
[refactor]거래내역 목록 조회 2번째 코드 개선

vayu404 commented:
/gemini review

vayu404 commented on src/main/java/dev/woori/wooriLearn/domain/edubankapi/eduaccount/service/EdubankapiAccountService.java:
```diff
@@ -0,0 +1,93 @@
+    /**
+     *   거래내역 목록 조회
+     */
+    public List<EdubankapiTransactionHistoryDto> getTransactionList(
+            Long accountId,
+            String period,
+            LocalDate startDate,
+            LocalDate endDate,
+            String type
+    ) {
```
이 부분은 다른 기능들 개발 후에 

고도화 작업에서 해도 좋을 것 같습니다

vayu404 commented on src/main/java/dev/woori/wooriLearn/domain/edubankapi/eduaccount/service/EdubankapiAccountService.java:
```diff
@@ -0,0 +1,93 @@
+                    if ("DEPOSIT".equalsIgnoreCase(type)) return h.getAmount() > 0;
+                    if ("WITHDRAW".equalsIgnoreCase(type)) return h.getAmount() < 0;
+                    return false;           // all, deposit, withdraw 이외의 값이 들어오면 예외처리
+                })
```
교육용 거래내역 목록 조회이기 때문에 

거래 내역이 많이 없어서 기존 코드 유지하겠습니다

Commit:
[refactor]거래내역 목록 조회 3번째 코드 개선

kimhm0828 reviewed (approved):
수고하셨습니다~ 나중에 페이지네이션 이용해서 개선하면 좋을 것 같아요
충돌 해결한 뒤에 머지해 주시면 됩니다!

kimhm0828 committed:
[refactor] 브랜치 병합에 따른 충돌 수정

# Conflicts:
#	src/main/java/dev/woori/wooriLearn/domain/account/service/AccountService.java
